test/supplemental_genbank_tests/genbank_upload_full_test.py

```diff
@@ -152,18 +192,340 @@ def test_simple_shock_upload(self):
+            data.pop(key)
+        for key in ["cdss", "features", "mrnas", "non_coding_features"]:
+            for dist in data.get(key):
+                dist.pop("aliases", None)
+        for ontology_event in data.get("ontology_events", []):
+            ontology_event.pop("timestamp")
+            ontology_event.pop("ontology_ref")
+        return data
+
+    def _retrieve_assembly_data(self, data):
+        # make a copy to avoid modifying the original assembly data
+        data = dict(data)
+        data.pop("fasta_handle_ref")
+        fasta_handle_info = data["fasta_handle_info"]
+        fasta_handle_info.pop("shock_id")
+        for key in ["hid", "id"]:
+            fasta_handle_info["handle"].pop(key)
+        return data
+
+    def _get_object(self, result, is_genome):
+        ref = 'genome_ref' if is_genome else 'assembly_ref'
+        assert _OBJECT_VERSION_PATTERN.match(result[ref])
+        return self.wsClient.get_objects2({"objects": [{'ref': result[ref]}]})["data"][0]
+
+    def _check_info(self, obj, result, file_names, idx, expected_metadata, is_genome):
+        info = obj["info"]
+        object_info = 'genome_info' if is_genome else 'assembly_info'
+        object_name = file_names[idx] if is_genome else file_names[idx] + "_assembly"
+        object_type = 'KBaseGenomes.Genome' if is_genome else 'KBaseGenomeAnnotations.Assembly'
+        retrieved_metadata = self._retrieve_genome_metadata(info[10]) if is_genome else info[10]
+
+        assert info == result[object_info]
+        assert info[1] == object_name
+        assert info[2].split('-')[0] == object_type
+
+        # datetime.fromisoformat is not available in Python 3.6 or below
+        assert datetime.strptime(info[3], '%Y-%m-%dT%H:%M:%S+%f')
+        assert info[6] == self.wsID
+        assert info[7] == self.wsName
+
+        # check metadata
+        assert retrieved_metadata == expected_metadata[idx]
+
+    def _check_prov(self, obj, expected_provenance):
+        provenance = obj["provenance"]
+        retrieved_provenance = self._retrieve_provenance(provenance)
+        assert retrieved_provenance == expected_provenance
+
+    def _check_data(self, obj, idx, expected_data, is_genome):
+        data = obj["data"]
+        retrieved_data = (
+            self._retrieve_genome_data(data)
+            if is_genome
+            else self._retrieve_assembly_data(data)
+        )
+        assert retrieved_data == expected_data[idx]
+
+    def _check_result_object_info_provenance_data(
+        self,
+        results,
+        file_names,
+        expected_metadata,
+        expected_provenance,
+        expected_data,
+        is_genome=True,
+    ):
+        for idx, res in enumerate(results):
+            obj = self._get_object(res, is_genome)
+            self._check_info(obj, res, file_names, idx, expected_metadata, is_genome)
+            self._check_prov(obj, expected_provenance)
+            self._check_data(obj, idx, expected_data, is_genome)
```

Comment:
There's one more missing piece - the files in the blobstore need to be checked. This could be just pulling the blobstore node and checking the MD5 is what you expect for the file

Reply:
If I have done genome/assembly md5sum check above, anything else I need address for this conversation?

Reply:
If you've checked the md5 for all files that are uploaded to the blobstore as part of the import, then no

Reply:
👍